### DISCUSSION TITLE
Update active Wezterm window with CLI

### DISCUSSION THREAD

**Discussion by joshmedeski** [wezterm/wezterm#4300] [Ideas]
Hello everyone,

I would like to be able to update the active Wezterm window using the command line. I want to update the background and color scheme based on when I switch tmux sessions.

I will be writing some bash for my [t - the smart tmux session manager](https://github.com/joshmedeski/t-smart-tmux-session-manager) plugin.

Right now, I've been able to map a keybinding to reload the config and write the appropriate lua code to change the background. I want a way to find a way to update wezterm via a CLI command.

Perhaps we can add an `--active` page and it will allow me to update the current config of the active wezterm window.

```
$ wezterm --active --config enable_scroll_bar=true
```

My use case will be grabbing the photo from the root of current tmux directory (after switching) and overwriting the background config to switch to that image. There will be some rules to decide which image to use, and what to do when no image is found.

Can anyone help me figure out what I need to do to get this working?

Thanks!

p.s. big fan of Wezterm, thanks for all the hard work so far!

**Comment by wez** [comment]
There isn't a way to achieve this directly.
You can use user vars to pass data and trigger an event, and in that event set config overrides, if you have some lua that cooperates with the application you are building.

https://wezfurlong.org/wezterm/recipes/passing-data.html#user-vars for the general concept

https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html for overriding config

**Comment by joshmedeski** [comment]
Thanks, that's exactly what I need to mess around with my program.
